support/jwt/src/main/kotlin/org/doorip/support/jwt/JwtProvider.kt

```
@@ -0,0 +1,87 @@
+
+    fun parseHeader(token: String): Map<String, String>? {
+        try {
+            val header = token.split(".")[0]
+            val decoder = Base64.getUrlDecoder()
+            val decoded = String(decoder.decode(header))
+            return objectMapper.readValue(decoded, typeReference)
+        } catch (e: DatabindException) {
+            return null
+        } catch (e: JsonParseException) {
+            return null
+        } catch (e: IndexOutOfBoundsException) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)